cloudbuild.yaml

```diff
@@ -1,26 +1,19 @@
-#############################################################################
-## The top section of this file is identical in the 3 cloudbuild.*yaml files.
-## Make sure any edits you make here are copied over to the other files too
-## if appropriate.
-##
-## TODO(al): consider if it's possible to merge these 3 files and control via
-## substitutions.
-#############################################################################
+# This file contains Google Cloud Build configuration for presubmit checks, unit
+# and integration tests, triggered by pull requests and commits to branches.
 
 timeout: 1800s
 substitutions:
   _CODECOV_TOKEN: "" # The auth token for uploading coverage to Codecov.
 options:
   machineType: E2_HIGHCPU_32
   volumes:
-  # A shared volume for caching Go modules between steps.
-  - name: go-modules
+  # A shared volume for caching Go modules, binaries and builds between steps.
+  - name: go-workspace
     path: /go
   env:
     - GO111MODULE=on
-    - GOPROXY=https://proxy.golang.org
-    - PROJECT_ROOT=github.com/google/trillian
     - GOPATH=/go
+    - GOCACHE=/go/.cache/go-build
```

Comment:
Not sure this should be added. It doesn't seem to bring a big speed-up. But there is a risk that it caches some succeeded tests, and prevent running them in another step. The doc says it shouldn't happen if build flags differ, but it doesn't cover all cases (e.g. the `cgo` changes).

Reply:
Removed.